content/testable-architecture.md

```diff
@@ -3,4 +3,62 @@ title: Testable Architecture
 description: Designing for testability
 ---
 
-Testable Architecture
\ No newline at end of file
+Automated tests and executable specifications can only work with software that has a testable architecture.
+A testable architecture enables fast and reliable tests that are easy to write, execute and maintain.
+Fast tests enable the developers to run them and obtain fast feedback on what they are building, without losing focus or flow.
+
+When designing for testability, make sure that your products and services are composed of loosely-coupled,
+well-encapsulated components or modules; following the principles of
+[SOLID design](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)).
+
+Decoupling your business logic from your infrastructure (the often slow and brittle components),
+makes it possible to test at different levels, with maximum confidence and minimum cost.
+
+* Ensure that you can run your tests underneath your UI. Do not depend solely on UI tests. They are slow, expensive, and hard to fix.
```

Comment:
I would replace 'underneath' with 'below'.
I would also not use the abbrevation UI. I would introduce it as
... below your user interface, UI.

I would add brittle to the list of properties for a UI test. 

They are slow, brittle, expensive, and hard to fix.